xled/auth.py

```diff
@@ -396,10 +403,17 @@ def challenge_response_valid(self, hw_address=None):
         expected = xled.security.make_challenge_response(self._challenge, hw_address)
         if expected != self._challenge_response:
             msg = (
-                "challenge-response invalid. "
+                "Generation 1 challenge-response invalid. "
+                "Received challenge-response: %r but %r was expected."
+            )
+            log.warning(msg, self._challenge_response, expected)
+            #raise ValidationError()
+        elif self._challenge != self._challenge_response:
```

Comment:
This is an `elif` - this means that this block will only be checked if the Generation 1 challenge was correct.

The reason this code allows the authentication process to work is not because `self._challenge == self._challenge_response` - that block is never executed. It's working because the first if no longer has a ValidationError, so this PR has the effect of ignoring any validation errors.

Reply:
Some debug I added locally:

```
log.debug("Mike2, %s, %s, %s", self._challenge, self._challenge_response, self._challenge != self._challenge_response)


DEBUG:xled.auth:Mike2, b'\xab9\x0b\x81^\x14\xa2\xe8\xda9<\xb5"N\x07\xca\xef\xe8y\rq\xa5\xb92\xc0\x83\xa7\xbb\xd1\xf2\xd5\x19', 85fab9352797960ee7ce2624ba9aef3273eb69a2, True
```

Also - these two fields are of different types (bytes vs string) and so will never be equal

Reply:
I think this PR would actually have the effect of breaking auth for v1 devices, as they will succeed the check, which means the if falls through (because this is an `if ERROR-CONDITION`) - and the second branch will always fail with a ValidationError